django/contrib/admin/utils.py

```diff
@@ -416,7 +416,7 @@ def display_for_value(value, empty_value_display, boolean=False):
 
     if boolean:
         return _boolean_icon(value)
-    elif value is None:
+    elif value is None or isinstance(value, str) and value == '':
```

Comment:
Could we just use `elif value is None or value == ''` here? As I can see, they have the same result.

Reply:
Do Django models support redefining of compairing operators (like __eq__). If so, in some cases it would be useful to check if value is string. But I'm not sure, if this way is suitable for such situations

Reply:
I don't understand at all why, but if we delete check if the value is str or not, there is one test in admin_widgets that is not successful

Reply:
It's because in AdminFileWidgetTests.test_readonly_fields, there is a FileField, which will produce '-' if we do't  check the value is str or not, but the unit test need empty string.
So, maybe the check here is necessary?